### PR TITLE
Add tempest-log-test image

### PR DESCRIPTION
**Pull request by kornicameister**

### CODE REVIEW

**Comment by kornicameister** [comment]
@timothyb89 I've been working on this topic before my holidays. Now getting back to it.
Hmm...I've started from rethinking the topic and kind of though if going toward the separate image is good idea. Kind of it doesn't make any sense. I'd rather consider two following ideas:

* adding the log-api to the existing image, running the container with some sort of argument to execute just the subset of available tests cases
* better in my opinion, make two images (vyeah :1st_place_medal: ) but base each of them respectively on api and log-api images. I mean, at the moment, we do install the api as python package inside the tempest image. Fine enough but worst case scenario means that you might end up launching the tempests for the different commit than you;ve build monasca-api image that's container tempests communicate with.

**Comment by kornicameister** [comment]
@timothyb89 @witekest friendly reminder

File: tempest-log-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+ * `latest`: refers to the latest tempest-log-tests container
+
+Usage
+-----
+
+To run, this container needs to connect to a working Keystone, ElasticSearch 
+and Monasca LOG API. It can be run using following command:
+
+```bash
+docker run --link elasticsearch --link keystone monasca/tempest-log-tests
```
Comment (matrixik) [inline tempest-log-tests/README.md]: Linking log api?

File: tempest-log-tests/start.sh [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+if [ "$AUTH_USE_SSL" = "true" ]; then
+  export AUTH_PROTOCOL="https"
+else
+  export AUTH_PROTOCOL="http"
+fi
+
+export AUTH_URI="${AUTH_PROTOCOL}://${KEYSTONE_SERVER}:${KEYSTONE_PORT}/v2.0/"
+export AUTH_URI_V3="${AUTH_PROTOCOL}://${KEYSTONE_SERVER}:${KEYSTONE_PORT}/v3/"
+
+if [ "$MONASCA_WAIT_FOR_API" = "true" ]; then
```
Comment (matrixik) [inline tempest-log-tests/start.sh]: `MONASCA_WAIT_FOR_LOG_API`

File: tempest-log-tests/start.sh [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  echo "Waiting for Monasca API to become available..."
+  success="false"
+
+  for i in $(seq $MONASCA_API_WAIT_RETRIES); do
```
Comment (matrixik) [inline tempest-log-tests/start.sh]: `MONASCA_LOG_API_WAIT_RETRIES`

File: tempest-log-tests/start.sh [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    monasca --os-user-domain-name "${OS_DOMAIN_NAME}" --os-project-name "${OS_TENANT_NAME}" \
+       --os-auth-url "${AUTH_URI_V3}" --os-username "${OS_USERNAME}" \
+       --os-password "${OS_PASSWORD}" alarm-list --limit 1
+    if [ $? -eq 0 ]; then
+      success="true"
+      break
+    else
+      echo "Monasca API not yet ready (attempt $i of $MONASCA_API_WAIT_RETRIES)"
```
Comment (matrixik) [inline tempest-log-tests/start.sh]: `MONASCA_LOG_API_WAIT_RETRIES`

File: tempest-log-tests/start.sh [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+      sleep "$MONASCA_API_WAIT_DELAY"
```
Comment (matrixik) [inline tempest-log-tests/start.sh]: `MONASCA_LOG_API_WAIT_DELAY`

File: tempest-log-tests/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ARG REPO=https://github.com/openstack/monasca-log-api
+ARG BRANCH="master"
+ARG UPPER_CONSTRAINTS=https://git.openstack.org/cgit/openstack/requirements/plain/upper-constraints.txt
+
+ENV GIT_SSL_NO_VERIFY true
+ENV no_proxy=127.0.0.1,localhost,localaddress,.localdomain.com,keystone,log-api,elasticsearch
+
+RUN apk add --no-cache python py2-pip git && \
+  apk add --no-cache --virtual build-dep git python-dev make g++ linux-headers libxml2-dev libxslt-dev \
+  openssl-dev libffi-dev py-mysqldb
```
Comment (matrixik) [inline tempest-log-tests/Dockerfile]: I think you also need to install `ca-certificates`

**Comment by kornicameister** [comment]
@matrixik closing this on my own. Whenever needed you can reopen it.

**Pull request closed without merging**